browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -2539,6 +2555,59 @@ L.CanvasTileLayer = L.Layer.extend({
 		target.onclick = target.ontouchend = func;
 	},
 
+	_showFunctionListPopUp: function(position, textMsg) {
+		var parent = L.DomUtil.create('div', '');
+		L.DomUtil.createWithId('div', 'functionlist-pop-up-preview', parent);
+		var text = L.DomUtil.createWithId('p', 'functionlist-pop-up', parent);
+
+		var maxSuggestion = 3;
+		var functionNameList = [];
+		var resultText = '';
+		var currentFuncDescription = '';
+
+		var suggestionArray = JSON.parse(textMsg);
+		if (suggestionArray.length < maxSuggestion) { maxSuggestion = suggestionArray.length; }
+
+		for (var i = 0; i < maxSuggestion; i++) {
+			if (i == 0)
+				currentFuncDescription = suggestionArray[i].description;
+
+			var signature = suggestionArray[i].signature;
+			functionNameList.push(signature.substring(0,signature.indexOf('(')));
+		}
+
+		for (var i = 0; i < maxSuggestion; i++) {
+			if (i == 0)
+				resultText = resultText + '[' + functionNameList[i] + ']';
+			else
+				resultText = resultText + ', ' + functionNameList[i];
+		}
+
+		var remainingFuncCount = suggestionArray.length - maxSuggestion;
+		if (remainingFuncCount > 0)
+			resultText = resultText + ' and ' + remainingFuncCount + ' more';
+
+		resultText = resultText + ' : ' + currentFuncDescription;
+
+		text.innerText = resultText;
+
+		this._map.functionListPopup = new L.Popup({className: 'functionlist-popup', closeButton: false, closeOnClick: false, autoPan: false})
```

Comment:
We are trying to get rid of Leaflet. We should use own component, not `L.Popup`.
Could you try `showDocumentTooltip` created by Mike? https://github.com/CollaboraOnline/online/blob/0431c584653d49a66d3b85130ab8aa32e1374517/browser/src/control/Control.UIManager.js#L1025